src/vs/workbench/browser/parts/editor/multiRowEditorTabsControl.ts

```diff
@@ -0,0 +1,172 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import { Dimension } from 'vs/base/browser/dom';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { IEditorGroupsAccessor, IEditorGroupView } from 'vs/workbench/browser/parts/editor/editor';
+import { IEditorTabsControl } from 'vs/workbench/browser/parts/editor/editorTabsControl';
+import { MultiEditorTabsControl } from 'vs/workbench/browser/parts/editor/multiEditorTabsControl';
+import { IEditorPartOptions } from 'vs/workbench/common/editor';
+import { EditorInput } from 'vs/workbench/common/editor/editorInput';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { StickyEditorGroupModel, UnstickyEditorGroupModel } from 'vs/workbench/common/editor/filteredEditorGroup';
+import { IEditorTitleControlDimensions } from 'vs/workbench/browser/parts/editor/editorTitleControl';
+import { IReadonlyEditorGroupModel } from 'vs/workbench/common/editor/editorGroupModel';
+
+export class MultiRowEditorControl extends Disposable implements IEditorTabsControl {
+
+	private readonly stickyEditorTabsControl: IEditorTabsControl;
+	private readonly unstickyEditorTabsControl: IEditorTabsControl;
+
+	constructor(
+		private parent: HTMLElement,
+		private accessor: IEditorGroupsAccessor,
+		private group: IEditorGroupView,
+		private model: IReadonlyEditorGroupModel,
+		@IInstantiationService protected instantiationService: IInstantiationService
+	) {
+		super();
+
+		this.stickyEditorTabsControl = this._register(this.instantiationService.createInstance(MultiEditorTabsControl, this.parent, this.accessor, this.group, new StickyEditorGroupModel(this.model)));
+		this.unstickyEditorTabsControl = this._register(this.instantiationService.createInstance(MultiEditorTabsControl, this.parent, this.accessor, this.group, new UnstickyEditorGroupModel(this.model)));
+
+		this.handlePinnedTabsSeparateRowToolbars();
+	}
+
+	private handlePinnedTabsSeparateRowToolbars(): void {
+		if (this.group.count === 0) {
+			// Do nothing as no tab bar is visible
+			return;
+		}
+		// Ensure action toolbar is only visible once
+		if (this.group.count === this.group.stickyCount) {
+			this.parent.classList.toggle('second-tabbar-visible', false);
+		} else {
+			this.parent.classList.toggle('second-tabbar-visible', true);
+		}
+	}
+
+	private getEditorTabsController(editor: EditorInput): IEditorTabsControl {
+		return this.model.isSticky(editor) ? this.stickyEditorTabsControl : this.unstickyEditorTabsControl;
+	}
+
+	openEditor(editor: EditorInput): boolean {
+		const [editorTabController, otherTabController] = this.model.isSticky(editor) ? [this.stickyEditorTabsControl, this.unstickyEditorTabsControl] : [this.unstickyEditorTabsControl, this.stickyEditorTabsControl];
+		const didChange = editorTabController.openEditor(editor);
+		if (didChange) {
+			// HACK: To render all editor tabs on startup, otherwise only one row gets rendered
```

Comment:
Why is that?

Reply:
At the time of initializing the EditorTabControl on startup, the editors have not yet been initialized, causing 0 tabs to be drawn. When the editors have been loaded, `openEditor` is called which triggers the drawing of all pinned tabs in the EditorTabControl. If we forward the `openEditor` call only to the tab bar the editor belongs to, the other tab bar will not be drawn.